### PR TITLE
Allowing for null values when editing optional offer fields

### LINKED ISSUE #260: Unable to remove optional offer fields
This discussion started in https://github.com/NIAEFEUP/nijobs-fe/issues/250 and we discovered the backend also needs some changing regarding this. Basically, some of the offer optional fields should be removable (e.g. `applyURL` and `vacancies`). The companies might want to remove them from their offers and.

Currently, setting these values as `null` in the `/offers/edit` endpoint returns the following error:
image

This type of operation should be supported for the optional fields that make sense (without default values for example)

### PR DESCRIPTION
**Pull request by BrunoRosendo**
closes #260

### CODE REVIEW

**Comment by BrunoRosendo** [comment]
This PR does not include jobMinDuration and jobMaxDuration. These fields are weird to me, they're optional on the BE but mandatory on the FE form. Besides from that, the validations gets weird when editing because those fields and mandatory when the other is present

**Comment by Naapperas** [comment]
> This PR does not include jobMinDuration and jobMaxDuration. These fields are weird to me, they're optional on the BE but mandatory on the FE form. Besides that, the validations get weird when editing because those fields and mandatory when the other is present

After discussing this with @DoStini we concluded it is better to have both fields mandatory on the BE: it simplifies things in the long run. For the case when the `jobMaxDuration` equals 12, I think we should interpret 12 as being "12 or more months" to reflect the UI that is shown to the user on the FE.

File: test/end-to-end/offer.js [hunk 1]
```diff
@@ -2883,6 +2883,15 @@ describe("Offer endpoint tests", () => {
...
                     expect(res.body).toHaveProperty("applyURL", applyURL);
                 });
+
+                test("Should edit if applyURL is specified as null", async () => {
+                    const applyURL = null;
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({ applyURL }))
+                        .expect(HTTPStatus.OK);
+                    expect(res.body).toHaveProperty("applyURL", applyURL);
```
Comment (Naapperas) [inline test/end-to-end/offer.js:2893]: Just out of curiosity: why is this test different from the others?

**Review by Naapperas — APPROVED** [review]

**Review by AvilaAndre — APPROVED** [review]
I have little to no experience in Express but it looks good to me.

File: test/end-to-end/offer.js (hunk 1 above)
Comment (BrunoRosendo) [reply test/end-to-end/offer.js:2893]: The `applyURL` tests had that structure so I just followed it :shrug: Nothing that really matters

**Comment by BrunoRosendo** [comment]
> > This PR does not include jobMinDuration and jobMaxDuration. These fields are weird to me, they're optional on the BE but mandatory on the FE form. Besides that, the validations get weird when editing because those fields and mandatory when the other is present
> 
> After discussing this with @DoStini we concluded it is better to have both fields mandatory on the BE: it simplifies things in the long run. For the case when the `jobMaxDuration` equals 12, I think we should interpret 12 as being "12 or more months" to reflect the UI that is shown to the user on the FE.

Sounds good, I opened https://github.com/NIAEFEUP/nijobs-be/issues/263
Merging this now

**Pull request merged**